package slave;

option cc_generic_services = true;

message RunCommandRequest {
  required string command = 1;
  required uint32 edge_id = 2;

  //Directories necessary for outputs.
  repeated string dirs = 3;

  // Response file, if needed.
  optional string rspfile_name = 4;
  optional string rspfile_content = 5;
};

message RunCommandResponse {
  enum ExitStatus {
    kExitSuccess = 0;
    kExitFailure = 1;
    kExitInterrupted = 2;
  };

  required ExitStatus status = 1;
  required string output = 2;
  required uint32 edge_id = 3;
};

message FinishRequest {
};

message FinishResponse {
};

service SlaveService {
  rpc RunCommand(RunCommandRequest) returns (RunCommandResponse);
  rpc Finish(FinishRequest) returns (FinishResponse);
};
